rust/kernel/platdev.rs

```diff
@@ -85,5 +100,10 @@ impl Drop for Registration {
             // safe to call.
             unsafe { bindings::platform_driver_unregister(&mut self.pdrv) }
         }
+        if let Some(ptr) = self.of_table {
+            // SAFETY: `ptr` came from an `OfMatchTable`.
+            let tbl = unsafe { OfMatchTable::from_pointer(ptr) };
+            drop(tbl);
```

Comment:
I don't understand why we need to drop it

Reply:
The `drop(tbl)` is completely redundant, `tbl` would get dropped even without it. It's  just there to indicate to the reader that we are dropping the table there.

Reply:
I see. Imao, it's not pretty :sweat_smile:

Reply:
Hehe I agree... I think I used the word [ugly](https://github.com/Rust-for-Linux/linux/pull/276#issuecomment-844475911) myself :)